### ISSUE TITLE
Unused import inspection false positive in macros

### ISSUE THREAD

**Issue by mati865** [intellij-rust/intellij-rust#9853]
## Environment

* **IntelliJ Rust plugin version:** 0.4.184.5057-223
* **Rust toolchain version:** 1.67.0-nightly (e1d819583 2022-12-05) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.3 (CL-223.7571.171)
* **Operating system:** Windows 10 10.0
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description

Unused import inspection incorrectly reports imports as unused if the are used inside macros (both normal and proc).

## Steps to reproduce
Open this code as 2021 edition:
```rust
use std::f64::consts::PI;  // warning: Unused import: `std::f64::consts::PI`

fn foo() {
    println!("{PI}");
}
```

**Comment by neonaot** [comment]
Related to https://github.com/intellij-rust/intellij-rust/issues/7338

The problem is not entirely in macros, the plugin does not fully support the new feature of `format!` macros
<img width="313" alt="image" src="https://user-images.githubusercontent.com/49211026/205975608-9005e0e7-1313-4375-a96e-c4c5188906f8.png">

**Comment by mati865** [comment]
Example of proc macro with the issue:

`Cargo.toml`:
```toml
[package]
name = "a"
version = "0.1.0"
edition = "2021"

[dev-dependencies]
test-case = "2.2.2"
time = { version = "0.3.17", features = ["macros"] }
```
`src/lib.rs`:
```rust
#[cfg(test)]
mod test {
    use test_case::test_case;
    use time::macros::datetime;

    #[test_case(datetime!(2020-01-01 0:00 UTC))]
    fn test(_: time::OffsetDateTime) {}
}
```
Warning:
![image](https://user-images.githubusercontent.com/1174646/205993503-fe0fa427-bec1-486e-a8f8-fc38484b3e97.png)

**Comment by augusto-vaultree** [comment]
@dima74, hello, nice to meet you.

Are you working on this issue already? I'm quite bothered by this problem, as it generated quite a few false-positives, so was wondering when would it be fixed.
